jerry-core/jerry.c

```diff
@@ -519,6 +519,56 @@ jerry_value_is_undefined (const jerry_value_t value) /**< api value */
 } /* jerry_value_is_undefined */
 
 /**
+ * Check if the specified feature is enabled.
+ *
+ * @return true  - if the specified feature is enabled,
+ *         false - otherwise.
+ */
+bool jerry_is_feature_enabled (const jerry_feature_t feature)
+{
+  switch (feature)
+  {
+#ifdef JERRY_ENABLE_ERROR_MESSAGES
+    case JERRY_FEATURE_ERROR_MESSAGES:
+#endif /* JERRY_ENABLE_ERROR_MESSAGES */
+#ifdef JERRY_VALGRIND
+    case JERRY_FEATURE_VALGRIND:
+#endif /* JERRY_VALGRIND */
+#ifdef JERRY_VALGRIND_FREYA
+    case JERRY_FEATURE_VALGRIND_FREYA:
+#endif /* JERRY_VALGRIND_FREYA */
+#ifdef JERRY_CPOINTER_32_BIT
+    case JERRY_FEATURE_CPOINTER_32_BIT:
+#endif /* JERRY_CPOINTER_32_BIT */
+#ifdef JMEM_GC_BEFORE_EACH_ALLOC
+    case JERRY_FEATURE_MEM_STRESS_TEST:
+#endif /* JMEM_GC_BEFORE_EACH_ALLOC */
+#ifdef JMEM_STATS
+    case JERRY_FEATURE_MEM_STATS:
+#endif /* JMEM_STATS */
+#ifdef PARSER_DUMP_BYTE_CODE
+    case JERRY_FEATURE_PARSER_DUMP:
+#endif /* PARSER_DUMP_BYTE_CODE */
+#ifdef REGEXP_DUMP_BYTE_CODE
+    case JERRY_FEATURE_REGEXP_DUMP:
+#endif /* REGEXP_DUMP_BYTE_CODE */
+#ifdef JERRY_ENABLE_SNAPSHOT_SAVE
+    case JERRY_FEATURE_SNAPSHOT_SAVE:
+#endif /* JERRY_ENABLE_SNAPSHOT_SAVE */
+#ifdef JERRY_ENABLE_SNAPSHOT_EXEC
+    case JERRY_FEATURE_SNAPSHOT_EXEC:
+#endif /* JERRY_ENABLE_SNAPSHOT_EXEC */
+    {
```

Comment:
IMHO every case should have their own return statement, so the default case could handle it. Like:

```c
...
#ifdef JERRY_ENABLE_SNAPSHOT_SAVE
     case JERRY_FEATURE_SNAPSHOT_SAVE:
     {
       return true;
     }
#endif /* JERRY_ENABLE_SNAPSHOT_SAVE */
#ifdef JERRY_ENABLE_SNAPSHOT_EXEC
     case JERRY_FEATURE_SNAPSHOT_EXEC:
     {
       return true;
     }
#endif /* JERRY_ENABLE_SNAPSHOT_EXEC */
    default:
     {
       return false;
     }
```